src/api.rs

```diff
@@ -0,0 +1,421 @@
+        match input {
+            UnifiedAssetJob::Vtf { path } => {
+                context.queue(VtfConfig, path);
+            }
+            UnifiedAssetJob::Vmt { path } => {
+                context.queue(self.material_config, path);
+            }
+            UnifiedAssetJob::Mdl { path, config } => {
+                context.queue(config, path);
+            }
+            UnifiedAssetJob::Vmf { path, config } => {
+                // VMF files need special handling - read and parse first
+                if let Ok(bytes) = context.fs().read(&path) {
+                    if let Ok(vmf) = Vmf::from_bytes(&bytes) {
+                        context.queue(config, vmf);
+                    } else {
+                        error!("Failed to parse VMF file: {}", path);
+                    }
+                } else {
+                    error!("Failed to read VMF file: {}", path);
+                }
+            }
+        }
+        Ok(())
+    }
+}
+
+/// Enum representing different types of assets that can be imported
+#[derive(Debug, Clone)]
+pub enum AssetImportJob {
+    Vmf {
+        path: PathBuf,
+        config: VmfConfig<MaterialConfig>,
+    },
+    Mdl {
+        path: PathBuf,
+        config: MdlConfig<MaterialConfig>,
+    },
+    Vmt {
+        path: PathBuf,
+    },
+    Vtf {
+        path: PathBuf,
+    },
+}
+
+/// Python wrapper for parallel import builder
+#[pyclass(module = "plumber", name = "ApiImporter")]
+pub struct PyApiImporter {
+    material_config: MaterialConfig,
+    executor: Option<Executor<BlenderAssetHandler>>,
+    receiver: Receiver<Message>,
+    jobs: Vec<AssetImportJob>,
+    callback_obj: PyObject,
+}
+
+#[pymethods]
+impl PyApiImporter {
+    #[new]
+    #[args(file_system, callback_obj, threads_suggestion, kwargs = "**")]
+    fn new(
+        file_system: &PyFileSystem,
+        callback_obj: PyObject,
+        threads_suggestion: usize,
+        kwargs: Option<&PyDict>,
+    ) -> PyResult<Self> {
+        let start = Instant::now();
+        info!(
+            "opening file system of game `{}` for API import...",
+            file_system.file_system.name
+        );
+
+        let opened = file_system
+            .file_system
+            .open()
+            .map_err(|e| PyIOError::new_err(e.to_string()))?;
+
+        info!(
+            "file system opened in {:.2} s",
+            start.elapsed().as_secs_f32()
+        );
+
+        let mut settings = HandlerSettings::default();
+
+        if let Some(kwargs) = kwargs {
+            for (key, value) in kwargs {
+                if value.is_none() {
+                    continue;
+                }
+
+                match key.extract()? {
+                    "import_materials" => settings.material.import_materials = value.extract()?,
+                    "import_lights" => settings.import_lights = value.extract()?,
+                    "light_factor" => settings.light.light_factor = value.extract()?,
```

Comment:
Need to update the Python API parallel job builder that all of these settings need to be defined builder-wide, not just material settings. Also check that these settings aren't in the add_job functions there.